evident/effect_size.py

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) 2018, The Evident Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+# import joblib
+import pickle
+
+import pandas as pd
+import numpy as np
+
+from os.path import join, basename, exists
+
+from itertools import combinations
+from functools import partial
+from skbio import DistanceMatrix
+# from skbio.stats.distance import permanova
+from scipy.stats import mannwhitneyu, spearmanr
+
+
+def effect_size(mappings, alphas, betas, output, jobs, permutations,
+                alpha_method, overwrite, na_values):
+    # As we can have multiple mapping, alpha or files, we will construct a mfs
+    # dictionary with all the dataframes. Additionally, we will load the
+    # data_dictionary.csv file so we can use it to process the data
+    mappings = {f: pd.read_csv(f, sep='\t', dtype=str, na_values=na_values)
+                for f in mappings}
+    for m, mf in mappings.items():
+        mappings[m].set_index('#SampleID', inplace=True)
+    if betas:
+        betas = {f: DistanceMatrix.read(f) for f in betas}
+        print(
+            'maps: %d, betas: %d, cols: %s' % (
+                len(mappings), len(betas), [
+                    len(m.columns.values) for _, m in mappings.items()]))
+
+# with joblib.parallel.Parallel(n_jobs=jobs, verbose=100) as par:
+#     par(joblib.delayed(
+#         _process_column)(bf, c, fname, method)
+# for bf, c, fname, method in _generate_betas(
+#     betas, mappings, permutations, output))
+    else:
+        alphas = {f: pd.read_csv(f, sep='\t', dtype=str, na_values=na_values)
+                  for f in alphas}
+        for a, af in alphas.items():
+            alphas[a].set_index('#SampleID', inplace=True)
+
+        for af, c, fname, method in _generate_alphas(alphas, mappings,
+                                                     output, alpha_method,
+                                                     overwrite):
+            _process_column(af, c, fname, method)
+
+
+# def _beta(permutations, data, xvalues, yvalues):
+#     x_ids = list(xvalues.index.values)
+#     y_ids = list(yvalues.index.values)
+#     ids = x_ids + y_ids
+#     data_test = data.filter(ids)
+#     permanova_result = permanova(
+#         distance_matrix=data_test,
+#         # we can use use either x or y cause they are the same
+#         column=xvalues.name,
+#         grouping=pd.concat([xvalues, yvalues]).to_frame(),
+#         permutations=permutations).to_dict()
+#     xvals = list(
+#         data_test.filter(xvalues.index.values).to_series().dropna().values)
+#     yvals = list(
+#         data_test.filter(yvalues.index.values).to_series().dropna().values)
+#     return (permanova_result['p-value'], permanova_result['test statistic'],
+#             xvals, yvals)
+
+
+def _alpha(alpha_method, data, xvalues, yvalues):
```

Comment:
Please remove the alpha options (`spearmanr`) and leave only `mannwhitneyu`, this includes from the click options

Reply:
Note that this will impact most code in here, including saving the files.

Reply:
done.